### PR TITLE
Hostgroup resize adds new host to ambari cluster

### PR DESCRIPTION
**Pull request by duncangrant**
The resize effector of the hostgroup(a dynamic cluster) previously just added the hosts and they had to be manually added to the ambari cluster using the web interface.  Now the agents are added automatically and service components are installed.

### CODE REVIEW

File: ambari/src/main/java/io/brooklyn/ambari/EtcHostsManager.java [hunk 1]
```diff
@@ -117,7 +121,7 @@ public static void setHostsOnMachines(Iterable<? extends Entity> machines, Attri
      * Scans the entities, determines the IP address and hostname for each entity, and returns a map that connects from
      * IP address to fully-qualified domain name.
      *
-     * @param entities entities to search for name/IP information.
+     * @param entities      entities to search for name/IP information.
```
Comment (tbouron) [inline ambari/src/main/java/io/brooklyn/ambari/EtcHostsManager.java]: Extra spaces (or tabulation)

File: ambari/src/main/java/io/brooklyn/ambari/hostgroup/AmbariHostGroupImpl.java [hunk 2]
```diff
@@ -75,8 +76,12 @@ public void init() {
     public Collection<Entity> resizeByDelta(int delta) {
         Collection<Entity> entities = super.resizeByDelta(delta);
 
-        if (delta != 0) {
+        if (delta > 0) {
```
Comment (tbouron) [inline ambari/src/main/java/io/brooklyn/ambari/hostgroup/AmbariHostGroupImpl.java:79]: How do we handle the reduction then?

File: ambari/src/main/java/io/brooklyn/ambari/rest/RequestCheckRunnable.java [hunk 3]
```diff
@@ -72,7 +74,8 @@ public Boolean call() throws Exception {
 
         if (!done) {
             throw new RuntimeException(builder.errorMessage);
-        };
+        }
+        ;
```
Comment (tbouron) [inline ambari/src/main/java/io/brooklyn/ambari/rest/RequestCheckRunnable.java]: Can be removed

File: ambari/src/main/java/io/brooklyn/ambari/hostgroup/AmbariHostGroupImpl.java (hunk 2 above)
Comment (duncangrant) [inline ambari/src/main/java/io/brooklyn/ambari/hostgroup/AmbariHostGroupImpl.java:79]: We don't.  I don't think I've done anything to make this worse. As far as ambari is concerned it just won't be able to find the removed servers (at least until the ip is recycled and then it just won't be able to access them).

**Comment by duncangrant** [comment]
@tbouron I've made the changes as an extra commit.  If you're happy let me know and I will squash before you merge.

File: ambari/src/main/java/io/brooklyn/ambari/hostgroup/AmbariHostGroupImpl.java (hunk 2 above)
Comment (tbouron) [inline ambari/src/main/java/io/brooklyn/ambari/hostgroup/AmbariHostGroupImpl.java:79]: Fair enough. The reduction can be done within another PR.

**Comment by tbouron** [comment]
LGTM. Can you squash the commits then? Thanks!

**Pull request merged**